lib/safira/roulette.ex

```diff
@@ -0,0 +1,199 @@
+    |> Prize.changeset(attrs)
+    |> Repo.insert()
+  end
+
+  @doc """
+  Updates a prize.
+
+  ## Examples
+
+      iex> update_prize(prize, %{field: new_value})
+      {:ok, %Prize{}}
+
+      iex> update_prize(prize, %{field: bad_value})
+      {:error, %Ecto.Changeset{}}
+
+  """
+  def update_prize(%Prize{} = prize, attrs) do
+    prize
+    |> Prize.changeset(attrs)
+    |> Repo.update()
+  end
+
+  @doc """
+  Deletes a Prize.
+
+  ## Examples
+
+      iex> delete_prize(prize)
+      {:ok, %Prize{}}
+
+      iex> delete_prize(prize)
+      {:error, %Ecto.Changeset{}}
+
+  """
+  def delete_prize(%Prize{} = prize) do
+    Repo.delete(prize)
+  end
+
+  @doc """
+  Returns an `%Ecto.Changeset{}` for tracking prize changes.
+
+  ## Examples
+
+      iex> change_prize(prize)
+      %Ecto.Changeset{source: %Prize{}}
+
+  """
+  def change_prize(%Prize{} = prize) do
+    Prize.changeset(prize, %{})
+  end
+
+  @doc """
+  Returns the list of attendees_prizes.
+
+  ## Examples
+
+      iex> list_attendees_prizes()
+      [%AttendeePrize{}, ...]
+
+  """
+  def list_attendees_prizes do
+    Repo.all(AttendeePrize)
+  end
+
+  @doc """
+  Gets a single attendee_prize.
+
+  Raises `Ecto.NoResultsError` if the Attendee prize does not exist.
+
+  ## Examples
+
+      iex> get_attendee_prize!(123)
+      %AttendeePrize{}
+
+      iex> get_attendee_prize!(456)
+      ** (Ecto.NoResultsError)
+
+  """
+  def get_attendee_prize!(id), do: Repo.get!(AttendeePrize, id)
+
+  @doc """
+  Creates a attendee_prize.
+
+  ## Examples
+
+      iex> create_attendee_prize(%{field: value})
+      {:ok, %AttendeePrize{}}
+
+      iex> create_attendee_prize(%{field: bad_value})
+      {:error, %Ecto.Changeset{}}
+
+  """
+  def create_attendee_prize(attrs \\ %{}) do
+    %AttendeePrize{}
+    |> AttendeePrize.changeset(attrs)
+    |> Repo.insert()
+  end
+
+  @doc """
+  Updates a attendee_prize.
+
+  ## Examples
+
+      iex> update_attendee_prize(attendee_prize, %{field: new_value})
+      {:ok, %AttendeePrize{}}
+
+      iex> update_attendee_prize(attendee_prize, %{field: bad_value})
+      {:error, %Ecto.Changeset{}}
+
+  """
+  def update_attendee_prize(%AttendeePrize{} = attendee_prize, attrs) do
+    attendee_prize
+    |> AttendeePrize.changeset(attrs)
+    |> Repo.update()
+  end
+
+  @doc """
+  Deletes a AttendeePrize.
+
+  ## Examples
+
+      iex> delete_attendee_prize(attendee_prize)
+      {:ok, %AttendeePrize{}}
+
+      iex> delete_attendee_prize(attendee_prize)
+      {:error, %Ecto.Changeset{}}
+
+  """
+  def delete_attendee_prize(%AttendeePrize{} = attendee_prize) do
+    Repo.delete(attendee_prize)
+  end
+
+  @doc """
+  Returns an `%Ecto.Changeset{}` for tracking attendee_prize changes.
+
+  ## Examples
+
+      iex> change_attendee_prize(attendee_prize)
+      %Ecto.Changeset{source: %AttendeePrize{}}
+
+  """
+  def change_attendee_prize(%AttendeePrize{} = attendee_prize) do
+    AttendeePrize.changeset(attendee_prize, %{})
+  end
```

Comment:
honestly just being extremely nitpicky here, but would there ever be any circumstance where we would need to change/delete a prize associated with an attendee?

Reply:
Same as above, this was automatically created, but I think we'll need this when we implement this part in the admin dashboard.